core/commands/storage/upload/upload/dc_repair_routing.go

```diff
@@ -0,0 +1,143 @@
+	},
+	Type: guardpb.RepairContract{},
+}
+
+var hostRepairResponseCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Host responds to repair requests.",
+		ShortDescription: `
+This command enquires the repairer with the contract, if agrees with the contract after negotiation,
+returns the repairer's signed contract to the invoker.`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("file-hash", true, false, "File hash for the host prepares to repair."),
+		cmds.StringArg("repair-shard-hash", true, false, "Shard hash for the host prepares to repair."),
+		cmds.StringArg("file-size", true, false, "Size of the repair file."),
+		cmds.StringArg("download-reward-amount", true, false, "Reward amount for download workload."),
+		cmds.StringArg("repair-reward-amount", true, false, "Reward amount for repair workload."),
+		cmds.StringArg("repair-pid", true, false, "Host Peer ID to send repair requests."),
+		cmds.StringArg("download_contract_id", true, false, " Contract ID associated with the download requests."),
+		cmds.StringArg("repair_contract_id", true, false, "Contract ID associated with the require requests."),
+	},
+	RunTimeout: 1 * time.Minute,
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		n, err := cmdenv.GetNode(env)
+		ctxParams, err := uh.ExtractContextParams(req, env)
+		repairId := req.Arguments[5]
+		hostId := ctxParams.N.Identity.Pretty()
+		if repairId != hostId {
+			return fmt.Errorf("repair id %s is mismatched with host id %s", repairId, hostId)
+		}
+		ns, err := helper.GetHostStorageConfig(ctxParams.Ctx, ctxParams.N)
+		var price uint64
+		if ns.RepairCustomizedPricing {
+			price = ns.RepairPriceCustomized
+		} else {
+			price = ns.RepairPriceDefault
+		}
+		requestPrice, err := strconv.ParseInt(req.Arguments[4], 10, 64)
+		if err != nil {
+			return nil
+		}
+		if price < uint64(requestPrice) {
+			return fmt.Errorf("host desired price %d is greater than request price %d", price, requestPrice)
+		}
+		fileHash := req.Arguments[0]
+		downloadContractId := req.Arguments[6]
+		repairContractId := req.Arguments[7]
+		lostShardHashes := strings.Split(req.Arguments[1], ",")
+		fileSize, err := strconv.ParseInt(req.Arguments[2], 10, 64)
+		downloadRewardAmout, err := strconv.ParseInt(req.Arguments[3], 10, 64)
+		repairRewardAmount, err := strconv.ParseInt(req.Arguments[4], 10, 64)
+		repaireReq := &guardpb.RepairContract{
+			FileHash:             fileHash,
+			LostShardHash:        lostShardHashes,
+			FileSize:             fileSize,
+			DownloadRewardAmount: downloadRewardAmout,
+			RepairRewardAmount:   repairRewardAmount,
+			RepairPid:            repairId,
+			RepairSignTime:       time.Now().UTC(),
+			DownloadContractId:   downloadContractId,
+			RepairContractId:     repairContractId,
+		}
+		sig, err := crypto.Sign(n.PrivateKey, repaireReq)
+		if err != nil {
+			return err
+		}
+		repaireReq.RepairSignature = sig
+		return cmds.EmitOnce(res, repaireReq)
```

Comment:
We met many timeout issue on using repair node to do broadcast, so this time, I suggest this prepare contract works in other go routine, so that this command will return receive msg or not, and contract will be send to guard using the guard grpc rpc.